### PR TITLE
Support for On-The-Fly GeoJson Layers and a new Mutator Transform

### PR DESCRIPTION
**Pull request by kevinkreiser**
The basic gist here is I want to add 2 features.

1. i want to be able to drop a new layer into a tile build and be able to reference that layer from queries. if the layer is not there on the filesystem i want the transforms that reference it to do nothing (same thing that happens when a layer isnt found). the bulk of this was accomplished over in: https://github.com/tilezen/tilequeue/pull/414
2. i want a new transform that allows me, with injected python, to be able to mutate what the properties or geometry of a given feature becomes. it seemed to me we already had some of these things especially for properties but geometry was lacking.

Once I had those two features implemented I added a landmarks layer to the queries which simply allows you to drop a geojson file into your tile build and mark buildings with it and become a point feature layer.

### CODE REVIEW

File: vectordatasource/transform.py
Comment (kevinkreiser) [inline vectordatasource/transform.py:1302]: this is moving over to `tilequeue` where the rest of the layers are loaded and ill add a new section to queries.yaml instead of having it be part of the transform

File: queries.yaml
Comment (kevinkreiser) [inline queries.yaml:349]: this property signifies an "inline" layer. when the features are loaded from file we expect them to be already processed. that is we will not try to load a `landmarks.yaml` and do processing on the data that is in the geojson file. we expect the tags and values to be there and the coordinates to already be in mercator projection

File: queries.yaml
Comment (kevinkreiser) [inline queries.yaml:555]: landmarks mark buildings with a superseded property so that when the two overlap the downstream consumer of the tiles will be able to tell

File: queries.yaml
Comment (kevinkreiser) [inline queries.yaml:564]: the final output of the landmarks layer is a point layer, basically landmark pois with a bunch of metadata about the landmark. the mutate transform lets us both modify the geom and the properties with python code that gets `eval`'d directly. pretty powerful stuff. you can reference existing `properties` of a feature using the `{properties}` tag and you can access the geometry using the `{geometry}` tag. in the future we might want to allow more stuff to be referenced.

File: queries.yaml
Comment (kevinkreiser) [inline queries.yaml:1895]: if a building gets marked by a landmark, we exclude it from merging with other buildings, keeping it a single entity

File: requirements.txt
Comment (kevinkreiser) [inline requirements.txt:18]: apparently my system thought i needed these, i didnt check to see if that was the case. i can revert

File: vectordatasource/transform.py
Comment (kevinkreiser) [inline vectordatasource/transform.py:9769]: pretty good explanation here. basically we loop over the features and eval the python in the queries.yaml to compute new properties or geometry. if you omitted one or the other (they are both optional) it just leaves the existing values unchanged.

File: queries.yaml
Comment (peitili) [reply queries.yaml:349]: Could we make the name more explicit about its behavior, like `pre_processed_layer_path` might be a better name?

File: integration-test/2095_inline_layers.py
Comment (peitili) [inline integration-test/2095_inline_layers.py:1]: did you check whether your PR introduces new test failures than the master branch?

File: vectordatasource/transform.py
Comment (peitili) [inline vectordatasource/transform.py:9776]: The comment is confusing here, if the assert is false, then both are missing, correct?

File: integration-test/2095_inline_layers.py
Comment (kevinkreiser) [reply integration-test/2095_inline_layers.py:1]: do we have a list of ones that are failing? i can check on my machine which ones fail and pass before and after my change and report back

File: vectordatasource/transform.py
Comment (kevinkreiser) [reply vectordatasource/transform.py:9776]: yes both are missing but you need at least one hence: "missing one of" as in you need at least one of them. i can reword it though

File: integration-test/2095_inline_layers.py
Comment (peitili) [reply integration-test/2095_inline_layers.py:1]: you can also check it on the CI report

Comment (kevinkreiser) [reply integration-test/2095_inline_layers.py:1]: the report is identical to masters last failed build, the tests fail at: `test.test_transform.BuildingsUnifyTest`

Comment (peitili) [reply integration-test/2095_inline_layers.py:1]: oh no, this is not good, it means none of integration tests got a chance to even run yet. I found the unit test failure was introduced recently https://github.com/tilezen/vector-datasource/pull/2093

@travisgrigsby would you help fix that unit test so that other integration tests can be run and checked?

Comment (travisgrigsby) [reply integration-test/2095_inline_layers.py:1]: done #2099

Comment (kevinkreiser) [reply integration-test/2095_inline_layers.py:1]: cool ill wait for that to merge, then merge master into this branch then compare the two outputs from CI

File: integration-test/2095_inline_layers.py
Comment (nvkelso) [inline integration-test/2095_inline_layers.py:43]: 🙇

File: queries.yaml
Comment (nvkelso) [inline queries.yaml:12]: Add docs

Comment (nvkelso) [reply queries.yaml:12]: See the traffic layers for inspiration for optional layer descriptions.

File: queries.yaml
Comment (nvkelso) [inline queries.yaml:561]: In all the places...

`superseded_by` instead of `superseded`.

Please use the same family of property names as WOF in Mapzen tradition: 

https://github.com/whosonfirst/whosonfirst-properties/blob/main/properties/wof/superseded_by.json

File: queries.yaml
Comment (kevinkreiser) [reply queries.yaml:12]: ha! i was looking around for docs, it seemed like even the changelog isnt up to date. ill grep around to see if i can find them!

File: queries.yaml
Comment (kevinkreiser) [reply queries.yaml:561]: ah wow i had no idea this existed, i'll swap it out

File: queries.yaml
Comment (nvkelso) [reply queries.yaml:12]: https://github.com/tilezen/vector-datasource/blob/master/docs/layers.md#layer-reference
and
https://github.com/tilezen/vector-datasource/blob/master/docs/layers.md#traffic-flow

Comment (nvkelso) [reply queries.yaml:12]: The CHANGELOG is out of date. Historically that's been updated as releases are tagged – and it's past time to tag v1.9. Thanks for the nudge.

File: vectordatasource/transform.py
Comment (peitili) [inline vectordatasource/transform.py:9789]: what is `max zoom` here?

Comment (kevinkreiser) [reply vectordatasource/transform.py:9789]: ah sorry the last zoom in the tile pyramids that we create (z16). there is code over in tilequeue here: https://github.com/tilezen/tilequeue/blob/master/tilequeue/process.py#L613 where it basically calls into the transforms a second time (hence re-entrancy)

File: integration-test/2095_inline_layers.py
Comment (kevinkreiser) [reply integration-test/2095_inline_layers.py:1]: i was finally able to get the build to run al the way and it has the same number of failures as master for integration tests:

master:
```
----------------------------------------------------------------------
Ran 2198 tests in 1000.156s

FAILED (failures=109)
```

this branch:
```
----------------------------------------------------------------------
Ran 2200 tests in 734.782s

FAILED (failures=109)
```

**Review by travisgrigsby — APPROVED** [review]

**Comment by travisgrigsby** [comment]
I'd remove the strikethrough in the description before merging

**Pull request merged**
